packages/imperative/src/config/cmd/convert-profiles/convert-profiles.handler.ts

```diff
@@ -0,0 +1,195 @@
+            try {
+                this.uninstallPlugin(pluginInfo.name);
+                params.response.console.log(`Removed obsolete plug-in: ${pluginInfo.name}`);
+            } catch (error) {
+                params.response.console.error(`Failed to uninstall plug-in "${pluginInfo.name}":\n    ${error}`);
+            }
+            if (pluginInfo.postUninstall) await pluginInfo.postUninstall();
+        }
+
+        if (oldProfileCount == 0) return;
+        await OverridesLoader.ensureCredentialManagerLoaded();
+
+        const convertResult = await ConfigBuilder.convert(profilesRootDir);
+        for (const [k, v] of Object.entries(convertResult.profilesConverted)) {
+            params.response.console.log(`Converted ${k} profiles: ${v.join(", ")}`);
+        }
+        if (convertResult.profilesFailed.length > 0) {
+            params.response.console.log("");
+            params.response.console.errorHeader(`Failed to convert ${convertResult.profilesFailed.length} profile(s). See details below`);
+            for (const { name, type, error } of convertResult.profilesFailed) {
+                if (name != null) {
+                    params.response.console.error(`Failed to load ${type} profile "${name}":\n    ${error}`);
+                } else {
+                    params.response.console.error(`Failed to find default ${type} profile:\n    ${error}`);
+                }
+            }
+        }
+
+        params.response.console.log("");
+        const teamConfig = ImperativeConfig.instance.config;
+        teamConfig.api.layers.activate(false, true);
+        teamConfig.api.layers.merge(convertResult.config);
+        ConfigSchema.updateSchema();
+        await teamConfig.save(false);
+
+        const oldProfilesDir = `${profilesRootDir.replace(/[\\/]$/, "")}-old`;
+        try {
+            fs.renameSync(profilesRootDir, oldProfilesDir);
+        } catch (error) {
+            params.response.console.error(`Failed to rename profiles directory to ${oldProfilesDir}:\n    ${error}`);
+        }
+
+        const cliBin = ImperativeConfig.instance.rootCommandName;
+        // TODO Implement config edit command
+        params.response.console.log(`Your new profiles have been saved to ${teamConfig.layerActive().path}.\n` +
+            `Run "${cliBin} config edit --global-config" to open this file in your default editor.\n\n` +
+            `Your old profiles have been moved to ${oldProfilesDir}.\n` +
+            `Run "${cliBin} config convert-profiles --delete" if you want to completely remove them.`);
+    }
+
+    /**
+     * Retrieve list of obsolete CLI plug-ins that should be uninstalled.
+     * @returns List of plugins to uninstall containing the following properties:
+     *  - `name` - Name of plugin to uninstall
+     *  - `preuninstall` - Optional callback to be run before uninstall
+     *  - `postuninstall` - Optional callback to be run after uninstall
+     */
+    private getObsoletePlugins(): IPluginToUninstall[] {
```

Comment:
Would an application developer care to know what are the plugins we consider obsolete ?

Reply:
Discussed in person and decided this is OK to leave private.